### PR TITLE
Editor Payout w/ Email

### PR DESCRIPTION
**Pull request by calvinhlee23**
### Code: 
- migrated Purchase App's models to be separated
- wrote script ```src/user/editor_payout_tasks.py```
- tested local
- cron job to run every system-time 3pm (pst?) 
- pulling rsc to usd
- keeping record of exchange rate
- Emailing w/ CSV receipts

### Test: 
- Migrations ran successfully
- Paid RSC: 
Screen Shot 2022-02-09 at 6 03 23 PM

- Email 
Screen Shot 2022-02-09 at 6 01 13 PM
- csv
Screen Shot 2022-02-09 at 6 01 17 PM

### CODE REVIEW

File: src/user/editor_payout_tasks.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    RscExchangeRate.objects.create(
+      rate=result['rate'],
+      target_currency=USD,
+    )
+
+    editors = User.objects.filter(
+        permissions__isnull=False,
+        permissions__access_type=EDITOR,
+        permissions__content_type=ContentType.objects.get_for_model(Hub)
+    ).distinct()
```
Comment (lightninglu10) [inline src/user/editor_payout_tasks.py:57]: Can we ignore our team editors here? Brian / Joyce / Kobe / Patrick / Leo / Calvin

File: src/user/editor_payout_tasks.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+from utils import sentry
+
+UNI_SWAP_BUNDLE_ID = 1  # their own hard-coded eth-bundle id
+UNI_SWAP_GRAPH_URI = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2"
+USD_PAY_AMOUNT_PER_MONTH = 3000
+USD_PER_RSC_PRICE_FLOOR = .033
+
+# TODO: (kobe) - API is under Calvin's name. Need to move to RH's account
+MORALIS_API_KEY = 'vlHzigIN9AYgxwTV2y55ruHrUYc08WsMFCTZNn4mUSLzJAdWMW5pCnUtrL0yqlwE'
+MORALIS_LOOKUP_URI = "https://deep-index.moralis.io/api/v2/erc20/{address}/price".format(address=WEB3_RSC_ADDRESS)
```
Comment (lightninglu10) [inline src/user/editor_payout_tasks.py:32]: I can sign us up for a team API key

File: src/user/editor_payout_tasks.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+from user.related_models.user_model import User
+from utils import sentry
+
+UNI_SWAP_BUNDLE_ID = 1  # their own hard-coded eth-bundle id
+UNI_SWAP_GRAPH_URI = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2"
+USD_PAY_AMOUNT_PER_MONTH = 3000
+USD_PER_RSC_PRICE_FLOOR = .033
+
+# TODO: (kobe) - API is under Calvin's name. Need to move to RH's account
+MORALIS_API_KEY = 'vlHzigIN9AYgxwTV2y55ruHrUYc08WsMFCTZNn4mUSLzJAdWMW5pCnUtrL0yqlwE'
```
[the remark this answers is not in the document]
Comment (calvinhlee23) [reply src/user/editor_payout_tasks.py:31]: yup. sounds good. I'll make sure to expire this key before merging to prod

File: src/researchhub/settings.py [hunk 4]
```diff
@@ -611,6 +611,18 @@ def before_send(event, hint):
         '0x2275736dfEf93a811Bb32156724C1FCF6FFd41be'
     )
 
+MORALIS_API_KEY = 'DsAnTYhpjnFmpwugvHFkyNg3WWdGFANANdGAEjMAsKDuZwZNeIsPs8Q4LGn0GGDL'
```
Comment (lightninglu10) [inline src/researchhub/settings.py:614]: os.environ.get('MORALIS_API_KEY', '')

Comment (calvinhlee23) [reply src/researchhub/settings.py]: updated

**Comment by calvinhlee23** [comment]
@yattias, Will do. Thanks for the comment!

**Review by lightninglu10 — APPROVED** [review]

File: src/user/editor_payout_tasks.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        try:
+            pay_amount = result['pay_amount']
+            distributor = Distributor(
+                  # this is NOT the model. It's a simple object
+                  Distribution('EDITOR_PAYOUT', pay_amount, False),
+                  editor,
+                  None,
+                  today
+              )
+            distributor.distribute()
```
Comment (lightninglu10) [inline src/user/editor_payout_tasks.py:79]: if you don't run distribute here -- or make it a feature flag then the payout will not happen.

**Pull request merged**